__app__/crop/ingress_weather.py

```diff
@@ -66,237 +39,133 @@ def import_zensie_weather_data(
     Returns:
         status, error
     """
-
-    log: str = ""
-    success: bool = False
-    engine: tuple = ()
-    sensor_type: str = CONST_ZENSIE_WEATHER_SENSOR_TYPE
-
+    # connect to the DB to get weather data already there, so we don't duplicate
     success, log, engine = connect_db(conn_string, database)
-
     if not success:
         logging.error(log)
         return success, log
+    session = session_open(engine)
+    df_db = get_db_weather_data(session, dt_from,dt_to)
 
-    # get the list of zensie weather sensors
-    try:
-        session = session_open(engine)
-        zensie_sensor_list: list = get_zensie_sensors_list(session, sensor_type)
-        session_close(session)
-        if zensie_sensor_list is None or len(zensie_sensor_list) == 0:
-            success = False
-            log = "No sensors with sensor type {} were found.".format(sensor_type)
-    except:
-        session_close(session)
-        success = False
-        log = "No sensors with sensor type {} were found.".format(sensor_type)
+    # now get the Openweathermap API data
+    success, error, df_api = get_openweathermap_data(dt_from, dt_to)
 
-    if not success:
-        logging.error(log)
-        return log_upload_event(
-            CONST_ZENSIE_WEATHER_SENSOR_TYPE,
-            "Zensie Weather API",
-            success,
-            log,
-            conn_string,
-        )
+    # filter out the rows that are already in the db data
+    new_data_df = df_api[~df_api.index.isin(df_db.index)]
 
-    for _, zensie_sensor in enumerate(zensie_sensor_list):
-        sensor_id = zensie_sensor["id"]
-        sensor_check_id = zensie_sensor["device_id"]
-        logging.info(
-            "sensor_id: {} | sensor_check_id: {}".format(sensor_id, sensor_check_id)
+    logging.info(
+        "new data with size len(new_data_df): {}\n\n".format(
+                        len(new_data_df)
         )
-        if sensor_id > 0 and len(sensor_check_id) > 0:
-            logging.info(
-                "\nUpdate: Getting data for sensor_id: {} | dt_from: {}, dt_to: {}".format(
-                    sensor_id, dt_from, dt_to
-                )
+    )
+    if len(new_data_df) > 0:
+        # this is the current time in seconds since epoch
+        start_time: float = time.time()
+        session = session_open(engine)
+        for idx, row in new_data_df.iterrows():
+            data = ReadingsWeatherClass(
+                sensor_id=0,
+                timestamp=idx,
+                temperature=row["temperature"],
+                rain_probability=0., # not in openweathermap data
```

Comment:
This and radiation could also be None/null?

Reply:
good point - this would indeed be better - making the change now.